src/Language/PureScript/Sugar.hs

```diff
@@ -65,14 +66,25 @@ import Language.PureScript.Sugar.TypeDeclarations as S
 --
 desugar :: (Applicative m, MonadSupply m, MonadError MultipleErrors m, MonadWriter MultipleErrors m) => [ExternsFile] -> [Module] -> m [Module]
 desugar externs =
+  fmap snd . desugarWithEnv externs
+
+-- |
+-- A variant of `desugar` which also returns the imports/exports Env.
+--
+desugarWithEnv :: (Applicative m, MonadSupply m, MonadError MultipleErrors m, MonadWriter MultipleErrors m) => [ExternsFile] -> [Module] -> m (Env, [Module])
+desugarWithEnv externs =
   map removeSignedLiterals
     >>> traverse desugarObjectConstructors
     >=> traverse desugarOperatorSections
     >=> traverse desugarDoModule
     >=> desugarCasesModule
     >=> desugarTypeDeclarationsModule
-    >=> desugarImports externs
-    >=> rebracket externs
-    >=> traverse deriveInstances
-    >=> desugarTypeClasses externs
-    >=> createBindingGroupsModule
+    >=> desugarImportsWithEnv externs
+    >=> withSecond
+        (   rebracket externs
+        >=> traverse deriveInstances
+        >=> desugarTypeClasses externs
+        >=> createBindingGroupsModule
+        )
+  where
+  withSecond f (x, y) = (x,) <$> f y
```

Comment:
Apparently not:

```
λ: let withSecond = (fmap . fmap) :: Functor f => (a -> f b) -> (z, a) -> f (z, b)

<interactive>:12:26:
    Couldn't match type ‘a1’ with ‘(z1, a1)’
      ‘a1’ is a rigid type variable bound by
           an expression type signature:
             Functor f1 => (a1 -> f1 b1) -> (z1, a1) -> f1 (z1, b1)
           at <interactive>:12:18
    Expected type: (a1 -> f1 b1) -> a1 -> (z1, b1)
      Actual type: (a1 -> b1) -> (z1, a1) -> (z1, b1)
    In the second argument of ‘(.)’, namely ‘fmap’
    In the expression:
        (fmap . fmap) :: Functor f => (a -> f b) -> (z, a) -> f (z, b)

<interactive>:12:26:
    Couldn't match type ‘b1’ with ‘f1 b1’
      ‘b1’ is a rigid type variable bound by
           an expression type signature:
             Functor f1 => (a1 -> f1 b1) -> (z1, a1) -> f1 (z1, b1)
           at <interactive>:12:18
    Expected type: (a1 -> f1 b1) -> a1 -> (z1, b1)
      Actual type: (a1 -> b1) -> (z1, a1) -> (z1, b1)
    In the second argument of ‘(.)’, namely ‘fmap’
    In the expression:
        (fmap . fmap) :: Functor f => (a -> f b) -> (z, a) -> f (z, b)
```

I'm not a fan of using the tuple instances in cases like these anyway, I think `\f (x, y) -> (x,) <$> f y` is always going to be clearer than anything using instances like `Functor ((,) a)`.